addon/storage-adapters/indexed-db.ts

```diff
@@ -0,0 +1,111 @@
+import { StorageAdapterInterface } from './storage-adapter';
+import EmberObject from '@ember/object';
+import { isPresent } from '@ember/utils';
+import Dexie from 'dexie';
+
+export const version = 1;
+export const tableName = 'logs';
+export const schema = {
+	[tableName]: '++_id'
+};
+
+export interface IndexedDbInterface extends StorageAdapterInterface {
+	database: string;
+}
+
+export default class IndexedDb extends EmberObject implements IndexedDbInterface {
+	public database!: string;
+	private db!: Dexie;
+	private table!: Dexie.Table<any, number>;
+
+	init() {
+		this._super(...arguments);
+
+		if (!this.database) {
+			throw new Error('IndexedDB storage adapter needs a database');
+		}
+
+		const db = new Dexie(this.database);
+
+		db.version(version).stores(schema);
+
+		this.db = db;
+
+		const table = db.table(tableName);
+
+		this.table = table;
+	}
+
+	async isSupported() {
+		return this.db.open().then(() => true, () => false);
+	}
+
+	count(this: IndexedDb) {
+		return this.table.count();
+	}
+
+	push(this: IndexedDb, ...items: any[]) {
+		return this.db.transaction('rw', this.table, () => {
```

Comment:
Why `transaction` on a single `bulkAdd` operation?

```suggestion
		return this.table.bulkAdd(items);
```